scripts/xonsh

```diff
@@ -1,3 +1,3 @@
-#!/usr/bin/env python -u
-from xonsh.main import main
-main()
+#!/usr/bin/env sh
+export PYTHONUNBUFFERED=1
+/usr/bin/env python -c "from xonsh.main import main; main()"
```

Comment:
The only difference I can think of is that in your version, the `xonsh.main.__name__ == "__main__"`, and in the version here `xonsh.main.__name__ == "/path/to/xonsh/main.py"` while `"__main__" ` is assigned to the `-c` string.  Not a meaningful distinction :)